### PR TITLE
Added initial PostgreSQL support for new deploys

### PR DESCRIPTION
**Pull request by devec0**
Add support for relating to PostgreSQL via the `db-admin` relation on the PostgreSQL charm.
TODO: Add action to migrate data from MySQL to PostgreSQL.

### CODE REVIEW

**Comment by ChrisMacNaughton** [comment]
Works for me!

```
Model   Controller  Cloud/Region         Version    SLA          Timestamp
gitlab  lxd         localhost/localhost  2.7-beta1  unsupported  10:15:49+02:00

App         Version         Status  Scale  Charm       Store       Rev  OS      Notes
gitlab      0.1-6-gda8e7ce  active      1  gitlab      local         0  ubuntu  
postgresql  10.9            active      1  postgresql  jujucharms  199  ubuntu  
redis       5.0.5           active      1  redis       jujucharms   22  ubuntu  

Unit           Workload  Agent  Machine  Public address  Ports     Message
gitlab/0*      active    idle   0        10.189.174.175            GitLab installed and configured
postgresql/0*  active    idle   1        10.189.174.229  5432/tcp  Live master (10.9)
redis/0*       active    idle   2        10.189.174.61   6379/tcp  Redis singleton available

Machine  State    DNS             Inst id        Series  AZ  Message
0        started  10.189.174.175  juju-c64c06-0  bionic      Running
1        started  10.189.174.229  juju-c64c06-1  bionic      Running
2        started  10.189.174.61   juju-c64c06-2  bionic      Running
```

**Comment by devec0** [comment]
I'm working on making sure the migration works prior to formally merging this, and this involves some changes to how the key/value store works, which will likely break deploys deployed from this branch if deployed prior to merging. Once I have the MySQL -> PostgreSQL migration script done, I'll merge and release this code, but the functional and unit tests are now passing (not that the unit test coverage is great on this charm) so I'm close to releasing this.

**Comment by devec0** [comment]
Migration and PostgreSQL functional tests have been added, as well as cleaning up the migration charm states.

File: actions.yaml [hunk 1]
```diff
@@ -1,2 +1,4 @@
 reconfigure:
   description: "Re-renders gitlab.rb configuration file from Juju state and runs `gitlab-ctl reconfigure`"
+migratedb:
```
Comment (ChrisMacNaughton) [inline actions.yaml]: In the readme, this is referenced as `migrate-db`

**Comment by devec0** [comment]
Ah, yep, I'm part way through renaming it to be consistent, as well as testing a live install to make sure the migration works.

I've picked up that GitLab needs some special handling if upgrading between major versions, so I'm putting some python-apt logic in place to check the installed package version, so to update to the latest same-major version prior to doing the upgrade to the next major. After which, I think this will be ready to merge.

**Comment by devec0** [comment]
OK, so I had the opportunity to test both an upgrade from 11.11.0 through 12.5 and a MySQL to PostgreSQL migration. Both work, however in my testing I did get my schema in a slightly confused state - which the upstream GitLab migrations don't handle. As a result of the experience, I've updated the README with some specific guidance on managing the migration using the tools exposed by the charm.

**Pull request merged**